### PR TITLE
Driver for Allo Katana DAC

### PR DESCRIPTION
**Pull request by allocom**
Allo Katana DAC uses Sabre ESS codec.

### CODE REVIEW

**Comment by clivem** [comment]
Have ESS changed their stance? AFAIK, there is still the issue for any of their programmable Sabre chips, (ie. I2C software controlled), that you sign a NDA to get the register info/address/values. 
Unless they have changed their stance, they have forbidden any Open Source driver where any register address or programming value, that is obtained from a NDA'd datasheet, being exposed in an Open Source driver! Maybe you know this, maybe you don't. Just a friendly warning before you potentially drag the RPi guys into a legal nightmare with ESS, for "publishing" such a driver......

**Comment by HiassofT** [comment]
The soundcard driver (allo-katana-dac.c) doesn't seem to do much, you can drop it and use the simple card or audio graph card driver instead.

Here's a (completely untested) dt overlay, please give that a try: https://gist.github.com/HiassofT/5f7aec2407f854e184aecdc7d502779e

**Comment by pelwell** [comment]
[ Comment 1 has been largely preempted before submission, but still stands. ]
1. I believe you may be violating your NDA by publishing this driver - do you really want to do this?
2. The codec is a standard component and its driver should be upstreamed.

**Comment by AlloKatana** [comment]
Hi,

 I am one of the designers of the Katana Dac (using ess9038q2m). We are very well aware of the NDA from ess sabre (that we signed). So we used the only solution possible. Katana Dac has an MCU on board with the actual protected information (nda) of Ess Sabre. The driver that we are submitting is the driver of the MCU (and the translation is done securely in the MCU). Let me know if you have any other questions.

**Comment by pelwell** [comment]
So is the codec a standard component or have you modified the MCU firmware for your application?

Is the API exposed by the MCU in the public domain?

**Comment by HiassofT** [comment]
If the codec driver (sabre-ess-i2c) isn't talking to the ESS chip but your MCU then you should name it differently - allo-ess-bridge-i2c or sth like that

**Comment by allocom** [comment]
Hi Pelwell, Codec is not a standard component. MCU firmware is custom written & will not be made public. Register addresses & its definition are not directly mapped to the ESS chip registers.

**Comment by pelwell** [comment]
In that case you should:
1. delete the sound card driver and use the overlay suggested by HiassofT, and
2. move the codec driver into sound/soc/bcm, giving it a product-specific name - allo_katana_codec.c would be fine.

**Comment by allocom** [comment]
I am checking the overlay suggested by HiassofT (Thanks HiassofT) & will move the codec driver.

**Comment by allocom** [comment]
Hello Pelwell, I modified to have only one file (allo_katana_codec.c) under bcm. Pls check if it is fine.

File: arch/arm/boot/dts/overlays/README [hunk 1]
```diff
@@ -314,6 +314,12 @@ Load:   dtoverlay=allo-digione
 Params: <None>
 
 
+Name:   allo-katana
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/README]: This should be `allo-katana-dac-audio`

File: arch/arm/boot/dts/overlays/README [hunk 2]
```diff
@@ -314,6 +314,12 @@ Load:   dtoverlay=allo-digione
...
+Info:   Configures the Allo Katana audio card
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/README]: `Allo Katana DAC audio card` for consistency? You can choose.

**Review by pelwell — COMMENTED** [review]
Apart from my README comments, that looks good.

**Comment by allocom** [comment]
Modified README file.

**Comment by AlloKatana** [comment]
Thx you Pelwell for merging, HiassofT for the suggestions and CliveM for the comment (valid).

**Comment by pelwell** [comment]
I used the wrong merge option - I intended to squash them - but I'll fix it up in the other kernel versions.

**Comment by allocom** [comment]
Thanks all

**Pull request merged**